src/components/document/document.tsx

```diff
@@ -441,24 +456,19 @@ export class DocumentComponent extends BaseComponent<IProps, IState> {
   }
 
   private handleNewDocumentClick = () => {
-    const docType = this.props.document.isLearningLog ? LearningLogDocument : PersonalDocument;
-    const docTypeString = this.props.document.isLearningLog ? "Learning Log" : "Personal Document";
-    const { appConfig: { defaultDocumentTitle } } = this.stores;
-    const nextTitle = this.stores.documents.getNextOtherDocumentTitle(this.stores.user, docType, defaultDocumentTitle);
-    this.stores.ui.prompt(`Name your new ${docTypeString}:`, `${nextTitle}`)
-      .then((title: string) => {
-        this.handleNewDocumentOpen(title)
-        .catch(this.stores.ui.setError);
-      });
+    const { document, onNewDocument } = this.props;
+    onNewDocument && onNewDocument(document);
   }
 
-  private handleNewDocumentOpen = async (title: string) => {
-    const { db } = this.stores;
-    const newDocType = this.props.document.isLearningLog ? LearningLogDocument : PersonalDocument;
-    const newDocument = await db.createOtherDocument(newDocType, {title});
-    if (newDocument) {
-      this.props.workspace.setAvailableDocument(newDocument);
-    }
+  private handleDocumentRename = () => {
+    const { document } = this.props;
+    const docTypeString = document.isPersonal ? "Personal Document" : "Learning Log";
+    this.stores.ui.prompt(`Rename your ${docTypeString}:`, document.title, `Renaming ${docTypeString}`)
+      .then((title: string) => {
+        if (title !== document.title) {
+          document.setTitle(title);
+        }
+      });
```

Comment:
Are there any characters that we need to strip out of name length requirements that need to be enforced?  I tried entering a very long document name, and it gets handled better than I expected.